extensions/sockets/src/main/java/org/mule/module/socket/internal/SocketUtils.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.internal;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.mule.runtime.core.util.Preconditions.checkArgument;
+import org.mule.module.socket.api.connection.AbstractSocketConnection;
+import org.mule.module.socket.api.exceptions.UnresolvableHostException;
+import org.mule.module.socket.api.socket.tcp.TcpSocketProperties;
+import org.mule.module.socket.api.socket.udp.UdpSocketProperties;
+import org.mule.module.socket.api.source.SocketAttributes;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionExceptionCode;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.api.message.NullPayload;
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.core.DefaultMuleMessage;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.serialization.ObjectSerializer;
+import org.mule.runtime.core.transformer.types.DataTypeFactory;
+import org.mule.runtime.core.util.IOUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketException;
+
+public class SocketUtils
```

Comment:
class should be final and have a private constructor so that nobody can instantiate it